auctions/english-auction/project/contracts/auction-contract/src/main.sw

```diff
@@ -29,72 +29,79 @@ storage {
     // This issue can be tracked here: https://github.com/FuelLabs/sway/issues/2465
     /// Stores the deposits made based on the user and auction ID.
     /// Map((user, auction id) => deposit)
-    deposits: StorageMap<(Identity, u64), AuctionAsset> = StorageMap {},
+    deposits: StorageMap<(Identity, u64), u64> = StorageMap {},
```

Comment:
Since we are refactoring how about the TODO above is included in this PR?

Reply:
What do you think of improving the internal structure but then instead of directly returning the Auction we fetch it from storage and then create a new type to return to the user without the mapping?

Reply:
Imo that would introduce unnecessary complexity